app/instance-initializers/ajax.js

```diff
@@ -0,0 +1,11 @@
+import AjaxPromise from "../utils/ajax-promise";
+
+export default {
+  name: "ajax",
+  initialize: function(app) {
+    const { container = app } = app;
+    const adapter = container.lookup("adapter:application");
+
+    AjaxPromise.setDefaultHeaders(() => adapter.get("headers"));
+  }
+};
```

Comment:
@swatijadhav This can go into the app you're trying to fix, rather than in the shared repo (that repo is dangerous)